src/gateways/googleAnalyticsClient.ts

```diff
@@ -110,6 +121,19 @@ class GoogleAnalyticsClient {
     this.flushPendingCommands();
   }
 
+  private referringHostName() {
```

Comment:
this logic probably shouldn't be in a gateway, lets move it to its own util or helper and give it its own unit tests

i can't decide if it would be clean to call that helper from in here, or pass the value into `setCustomDimensionForSession` as an argument, you can do whatever feels good to you